src/client/game.cpp

```diff
@@ -1251,7 +1251,8 @@ void Game::run()
 		updateCamera(dtime);
 		updateSound(dtime);
 		processPlayerInteraction(dtime, m_game_ui->m_flags.show_hud);
-		updateFrame(&graph, &stats, dtime, cam_view);
+		if (RenderingEngine::shouldRender())
+			updateFrame(&graph, &stats, dtime, cam_view);
```

Comment:
we have a problem here:
`updateFrame` for example calls `client->getParticleManager()->step(dtime);`

for all of the steps skipped the dtime will be lost.

Reply:
two possible fixes:
* carefully divide code between state keeping and actual drawing
* accumulate dtime and hope nothing breaks when unpausing

Reply:
> carefully divide code between state keeping and actual drawing

Turns out there is already a PR for that: #13967